src/globals.cpp

```diff
@@ -397,6 +402,26 @@ CTweak<uint64_t> grapheneFastFilterCompatibility("net.grapheneFastFilterCompatib
     "Support fast Bloom filter: 0 - either, 1 - fast only, 2 - regular only (default: either)",
     GRAPHENE_FAST_FILTER_SUPPORT);
 
+CTweak<bool> syncMempoolWithPeers("net.syncMempoolWithPeers", "Synchronize mempool with peers", false);
```

Comment:
This is a periodic sync, right?  Maybe to give extra configurability, change to an integer "mempoolSyncPeriod" in seconds with 0 meaning "off".

Reply:
Overall, I'm in favor of adding the flexibility you're proposing. One concern I have is in allowing the requester to dictate the update frequency. This could be an attack vector. Currently the requester will not allow syncing more often than every 25 seconds from any given peer. I could of course add a default allowable sync frequency for the responder and add a sync frequency negotiation method similar to version negotiation. Do you think that's the way to go?

For your second point, let me make sure I'm understanding correctly. You're saying that whatever the negotiated sync frequency *per node* should be spread out so that you're syncing as frequently as possible while respecting the sync frequency of each peer. Is that right? Currently we randomly choose a peer at the (universal) sync frequency. This balances load, but does not maximize throughput.